cypress/utils/apps/query_enhancements/saved.js

```diff
@@ -268,13 +268,37 @@ export const verifyDiscoverPageState = ({
   selectFields,
   sampleTableData,
 }) => {
-  cy.getElementByTestId('datasetSelectorButton').contains(dataset);
+  if (dataset) {
+    cy.getElementByTestId('datasetSelectorButton').contains(dataset);
+  }
+
+  // Escape special regex characters in the query string
+  const escapedQueryString = queryString.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+
+  // The Monaco editor renders spaces using various Unicode whitespace characters and middle dot characters
+  // This comprehensive pattern handles all possible whitespace representations that might appear in the editor
+  // including regular spaces, non-breaking spaces, middle dots, and other special characters used for spacing
+  const pattern = new RegExp(
+    escapedQueryString.replace(
+      /\s+/g,
+      '[\\s\\u00A0\\u00B7\\u2022\\u2023\\u25E6\\u2043\\u2219\\u22C5\\u30FB\\u00B7.·]+'
+    )
```

Comment:
Can we encapsulate this into a helper like `getSanitizedQueryString(queryString: string): string`. I don't know if we want the nitty gritty of sanitizing monaco text in query enhancements tests themself since this seems more like a monaco test helper.

Reply:
I will re-use `verifyDiscoverPageStateFromSaved` which is verifyDiscoverPageState from saved.ts to avoid duplication. All tests will just use verifyDiscoverPageState so we don't have to create a separate utility func for now. If we need to verify the string separately in the future use cases, then we will get this part out then.

Reply:
I'd suggest moving to a dedicated Monaco editor page object model so these modules have clear responsibilities. I don't feel `verifyDiscoverPageStateFromSaved` should have the responsibility for directly querying and parsing monaco's internal DOM alongside verifying high level product behavior for the discover page.

Reply:
I can move this to a separate helper func in autocomplete.js instead of keeping it in saved.js. Currently, we don't use  object model (OOM) for the tests. Instead we create one utility file for each major func/comp.